app/models/concerns/form526_claim_fast_tracking_concern.rb

```diff
@@ -132,6 +134,30 @@ def update_form_with_classification_code(classification_code)
     invalidate_form_hash
   end
 
+  def log_max_cfi_metrics_on_submit
+    if DISABILITIES_WITH_MAX_CFI.intersect?(diagnostic_codes)
+      DISABILITIES_WITH_MAX_CFI.each do |diagnostic_code|
+        selected_disability = disabilities.find do |dis|
+          diagnostic_code == dis['diagnosticCode']
+        end
+        if selected_disability.present?
+          rated_disabilities = rated_disabilities_from_ipf
+          rated_disability = rated_disabilities.find do |dis|
+            diagnostic_code == dis['diagnostic_code']
+          end
+
+          next if rated_disability.blank?
+
+          next unless rated_disability['maximum_rating_percentage'] == rated_disability['rating_percentage']
+
+          formatted_disability = selected_disability['name'].parameterize(separator: '_')
+          max_cfi_enabled = Flipper.enabled?(:disability_526_maximum_rating) ? 'on' : 'off'
+          StatsD.increment("#{MAX_CFI_STATSD_KEY_PREFIX}.#{max_cfi_enabled}.submit.#{formatted_disability}")
+        end
+      end
+    end
```

Comment:
I have a bunch of random ideas for how to tighten up this method a bit more; it's nicer than before but still has a fair bit of nesting.
- For starters, you could either convert the top-level `if` into an early return, or even just omit it.
- A more drastic idea: what if we just fetch the IPF every time and extract a list of maxed-out disabilities? The DB hit is pretty cheap, and it could save about half this effort.